### PR TITLE
translates tool names and title properties in schema

### PR DESCRIPTION
**Pull request by benib**
implements #2
this needs a Q server with https://github.com/nzzdev/Q-server/pull/43

### CODE REVIEW

File: client/src/pages/editor.js [hunk 1]
```diff
@@ -23,14 +23,40 @@ function getSchemaForSchemaEditor(schema) {
...
+@inject(ItemStore, MessageService, DialogService, I18N, EventAggregator)
 export class Editor {
 
-  constructor(itemStore, messageService, dialogService, i18n) {
+  constructor(itemStore, messageService, dialogService, i18n, ea) {
     this.itemStore = itemStore;
     this.messageService = messageService;
     this.dialogService = dialogService;
     this.i18n = i18n;
+    this.ea = ea;
```
Comment (fuenkchen) [inline client/src/pages/editor.js]: I would call it `eventAggregator`, so it's clear what's meant in code sections which are further below too.

File: client/src/pages/editor.js [hunk 2]
```diff
@@ -49,15 +75,23 @@ export class Editor {
...
-        return Promise.reject();
+        throw response;
       })
       .then(schema => {
         this.fullSchema = schema;
-        this.schema = getSchemaForSchemaEditor(schema);
-        if (schema.properties.hasOwnProperty('options')) {
-          this.optionsSchema = schema.properties.options;
+        const schemaForEditor = getSchemaForSchemaEditor(this.fullSchema);
+        this.schema = getTranslatedSchema(schemaForEditor, routeParams.tool, this.i18n);
```
Comment (fuenkchen) [inline client/src/pages/editor.js]: Could you put the following four lines of code to get the translated schema and options schema into a method and use it here and in the event bus?

File: client/src/pages/editor.js [hunk 3]
```diff
@@ -49,15 +75,23 @@ export class Editor {
...
+        if (this.fullSchema.properties.hasOwnProperty('options')) {
+          this.optionsSchema = getTranslatedSchema(this.fullSchema.properties.options, routeParams.tool, this.i18n);
         }
+
+        // whenever there is a language change, we calculate the schema and translate all title properties
+        this.ea.subscribe('i18n:locale:changed', () => {
+          this.schema = getTranslatedSchema(getSchemaForSchemaEditor(this.fullSchema), routeParams.tool, this.i18n);
```
Comment (fuenkchen) [inline client/src/pages/editor.js]: why don't you use `schemaForEditor` here?

**Comment by benib** [comment]
@fuenkchen agree with all all your comments 👍 and adressed them with in 2b39bb51301af663b5e1ea68fde78ddb4b21ad6e

**Pull request merged**